CrustTests/Company.swift

```diff
@@ -14,6 +14,35 @@ class Company {
 
 extension Company: AnyMappable { }
 
+enum CompanyCodingKey: Keypath {
+    case uuid
+    case employees(Set<EmployeeCodingKey>)
```

Comment:
The issue is the call to `map` must know all the keypaths it needs so there is a structure here that must exist at the top level. I have another idea that I think could add separation. I also make adding directives for codegen from MappingKeys easier.

```swift
enum Field<Key: MappingKey, NestedKey: MappingKey> { // MappingKey == Keypath
    .scalar(Key)
    .object(Key, NestedKey)
}
```
But this means having a new `KeyCollection` that can work with these structures. This seemed like a bunch of additional work and I just wanted to get done what we have done. Luckily, the way things are structured gives us a lot of flexibility to change syntax and behavior.